apicast/src/oauth/keycloak.lua

```diff
@@ -0,0 +1,243 @@
+local setmetatable = setmetatable
+local len = string.len
+local http_ng = require "resty.http_ng"
+local resty_url = require 'resty.url'
+local jwt = require 'resty.jwt'
+local cjson = require 'cjson'
+
+local _M = {
+  _VERSION = '0.1'
+}
+local mt = {
+  __index = _M
+}
+
+-- Required params for each grant type and response type.
+_M.params = {
+  grant_type = {
+    ['authorization_code'] = {'client_id','redirect_uri','code'},
+    ['password'] = {'client_id','client_secret','username','password'},
+    ['client_credentials'] = {'client_id','client_secret'}
+  },
+  response_type = {
+    ['code'] = {'client_id','redirect_uri'},
+    ['token'] = {'client_id','redirect_uri'}
+  }
+}
+
+function _M.init(endpoint)
+  _M.configured = endpoint
+
+  local config = { endpoint = endpoint}
+  if _M.configured then
+    _M.configuration = config
```

Comment:
We will have to revamp this to use the new configuration store and not using module level objects.